sycl/test-e2e/VirtualMem/memory_granularity.cpp

```diff
@@ -0,0 +1,76 @@
+// REQUIRES: aspect-ext_oneapi_virtual_mem
+
+// This test checks whether memory granularity returned is greater then 0 and
+// that the recommended granularity is the same size or greater than the minimum
+// granularity for virtul memory extension.
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <string_view>
+
+#include <sycl/detail/core.hpp>
+
+#include <sycl/ext/oneapi/virtual_mem/virtual_mem.hpp>
+
+namespace syclext = sycl::ext::oneapi::experimental;
+
+template <typename... ArgsT>
+int CheckReturnedGranularitySize(std::string_view TestName, ArgsT... Args) {
+  if (syclext::get_mem_granularity(Args...) == 0) {
+    std::cout << "Failed check: " << TestName << std::endl;
+    return 1;
+  }
+  return 0;
+}
+
+int main() {
+  int Failed = 0;
+  sycl::queue Q;
+  sycl::context Context = Q.get_context();
+  sycl::device Device = Q.get_device();
+
+  Failed += CheckReturnedGranularitySize(
+      "returned granularity size must be greater then 0 for device and "
+      "granularity_mode::recommended",
+      Device, Context, syclext::granularity_mode::recommended);
+  Failed += CheckReturnedGranularitySize(
+      "returned granularity size must be greater then 0 for device and "
+      "granularity_mode::minimum",
+      Device, Context, syclext::granularity_mode::minimum);
+  Failed += CheckReturnedGranularitySize(
+      "returned granularity size must be greater then 0 for context and "
+      "granularity_mode::recommended",
+      Context, syclext::granularity_mode::recommended);
+  Failed += CheckReturnedGranularitySize(
+      "returned granularity size must be greater then 0 for context and "
+      "granularity_mode::minimum",
+      Context, syclext::granularity_mode::minimum);
+
+  constexpr size_t NumberOfElements = 2;
+  std::array<size_t, NumberOfElements> RecommendedGranularities = {
+      syclext::get_mem_granularity(Device, Context,
+                                   syclext::granularity_mode::recommended),
+      syclext::get_mem_granularity(Context,
+                                   syclext::granularity_mode::recommended)};
+
+  std::array<size_t, NumberOfElements> MinimumGranularities = {
+      syclext::get_mem_granularity(Device, Context,
+                                   syclext::granularity_mode::minimum),
+      syclext::get_mem_granularity(Context,
+                                   syclext::granularity_mode::minimum)};
```

Comment:
Overall it seems like the whole test could be simplified to something like:

```c++
auto DevRecommended = syclext::get_mem_granularity(Device, Context,
                                   syclext::granularity_mode::recommended);
auto ContextRecommended = syclext::get_mem_granularity(Context,
                                   syclext::granularity_mode::recommended);
auto DevMinimum = syclext::get_mem_granularity(Device, Context,
                                   syclext::granularity_mode::minimum);
auto ContextMinimum = syclext::get_mem_granularity(Context,
                                   syclext::granularity_mode::minimum);
                                   
asssert(DevRecommended > 0 && "recommended granularity for device should be greater than 0");
asssert(ContextRecommended > 0 && "message");
// similar asserts for minimum
assert(DevRecommended >= DevMinimum && "message");
// similar assert for context
```

Reply:
Done